service-base/src/main/java/org/eclipse/hono/service/auth/device/CredentialsApiAuthProvider.java

```diff
@@ -206,47 +203,19 @@ public final void authenticate(
 
     @Override
     public final void authenticate(final JsonObject authInfo, final Handler<AsyncResult<User>> resultHandler) {
-        final ExecutionContext executionContext = new GenericExecutionContext(
-                TracingHelper.extractSpanContext(tracer, authInfo));
-        authenticate(authInfo, executionContext, s -> {
-            if (s.succeeded()) {
-                resultHandler.handle(Future.succeededFuture(s.result()));
-            } else {
-                resultHandler.handle(Future.failedFuture(s.cause()));
-            }
-        });
-    }
-
-    @Override
-    public final void authenticate(
-            final JsonObject authInfo,
-            final ExecutionContext executionContext,
-            final Handler<AsyncResult<DeviceUser>> resultHandler) {
 
-        Objects.requireNonNull(authInfo);
-        Objects.requireNonNull(executionContext);
-        Objects.requireNonNull(resultHandler);
-
-        final T credentials = getCredentials(authInfo);
+        final T credentials = getCredentials(Objects.requireNonNull(authInfo));
         if (credentials == null) {
             resultHandler.handle(Future.failedFuture(new ClientErrorException(HttpURLConnection.HTTP_UNAUTHORIZED, "malformed credentials")));
         } else {
-            authenticate(credentials, executionContext, resultHandler);
+            authenticate(credentials, TracingHelper.extractSpanContext(tracer, authInfo), s -> {
+                if (s.succeeded()) {
+                    resultHandler.handle(Future.succeededFuture(s.result()));
+                } else {
+                    resultHandler.handle(Future.failedFuture(s.cause()));
+                }
+            });
```

Comment:
`authenticate(credentials, TracingHelper.extractSpanContext(tracer, authInfo), resultHandler)` ?

Reply:
This won't work because of the generic AsyncResult type:
`resultHandler` is of type `Handler<AsyncResult<User>>` (coming from the vert.x `AuthProvider` interface, whereas the `DeviceCredentialsAuthProvider.authenticate` method expects the more specific `Handler<AsyncResult<DeviceUser>>`.

Reply:
I always get this wrong ;-)